microcosm/types.py

```diff
@@ -0,0 +1,8 @@
+from typing import Callable
```

Comment:
`microcosm.types` or `microcosm.typing`?  cc/ @adamhadani @afallou @cdm-ium

Reply:
I'm wondering if it makes sense to wire this up through `microcosm.api` so we can change it without breaking everything that depends on this. I don't care. @GabeLambda might... will ask him.

Reply:
I think I like the idea of putting these types in `microcosm.typing`, which is where we will keep things to be used for type annotations.  I don't think this restricts us too much.  Agreed @cdm-ium @GabeLambda ?